crates/networking/discv5/src/subnet.rs

```diff
@@ -0,0 +1,149 @@
+use alloy_rlp::{BufMut, Decodable, Encodable, bytes::Bytes};
+use anyhow::anyhow;
+use discv5::Enr;
+use ssz::{Decode, Encode};
+use ssz_derive::{Decode, Encode};
+use ssz_types::{BitVector, typenum::U64};
+use tracing::trace;
+
+pub const ATTESTATION_BITFIELD_ENR_KEY: &str = "attnets";
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum Subnet {
+    Attestation(u8),
+    SyncCommittee(u8),
+}
+
+#[derive(Clone, Debug, Default, Encode, Decode)]
+pub struct Subnets {
+    attestation_bits: Option<BitVector<U64>>,
+}
+
+impl Subnets {
+    pub fn new() -> Self {
+        Self {
+            attestation_bits: None,
+        }
+    }
+
+    pub fn enable_subnet(&mut self, subnet: Subnet) -> anyhow::Result<()> {
+        match subnet {
+            Subnet::Attestation(id) if id < 64 => {
+                let bits = self.attestation_bits.get_or_insert(BitVector::new());
+                bits.set(id as usize, true)
+                    .map_err(|_| anyhow!("Subnet ID out of bounds"))?;
```

Comment:
```suggestion
                    .map_err(|err| anyhow!("Subnet ID out of bounds: {err:?}"))?;
```

We might as well forward the error no?